filebeat/publish/publish.go

```diff
@@ -172,9 +172,12 @@ func (p *asyncLogPublisher) Start() {
 			case <-p.done:
 				return
 			case events := <-p.in:
-				pubEvents := make([]common.MapStr, len(events))
-				for i, event := range events {
-					pubEvents[i] = event.ToMapStr()
+
+				pubEvents := make([]common.MapStr, 0)
```

Comment:
`pubEvents := make([]common.MapStr, 0, len(events))`. See line 111. Consider helper function copying events and reuse in sync and async publisher modes.